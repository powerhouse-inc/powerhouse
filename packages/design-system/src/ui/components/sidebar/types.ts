import { type IconName } from "#powerhouse";

export enum NodeStatus {
  CREATED = "CREATED",
  MODIFIED = "MODIFIED",
  REMOVED = "REMOVED",
  MOVED = "MOVED",
  DUPLICATED = "DUPLICATED",
  UNCHANGED = "UNCHANGED", // default
}

export type SidebarIcon = IconName | React.ReactElement;

export type SidebarNode = {
  title: string;
  id: string;
  children?: SidebarNode[];
  icon?: SidebarIcon;
  expandedIcon?: SidebarIcon;
  status?: NodeStatus;
};

export interface FlattenedNode extends SidebarNode {
  depth: number;
  isExpanded: boolean;
}
