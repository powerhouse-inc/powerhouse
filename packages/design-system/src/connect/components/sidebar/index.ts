export * from "./sidebar";
export * from "./sidebar-footer";
export * from "./sidebar-header";
export * from "./sidebar-user";
