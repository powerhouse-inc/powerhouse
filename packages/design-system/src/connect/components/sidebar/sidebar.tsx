import { Sidebar, SidebarPanel, SidebarProps } from "@/powerhouse";
import {
  ConnectSidebarFooter,
  ConnectSidebarFooterProps,
} from "./sidebar-footer";
import {
  ConnectSidebarHeader,
  ConnectSidebarHeaderProps,
} from "./sidebar-header";

export interface ConnectSidebarProps
  extends Omit<SidebarProps, "maxWidth" | "minWidth">,
    ConnectSidebarHeaderProps,
    ConnectSidebarFooterProps {
  maxWidth?: string;
  minWidth?: string;
  headerContent?: React.ReactNode;
  loadingUser?: boolean;
  onLogin: () => void;
}

export const ConnectSidebar: React.FC<ConnectSidebarProps> = ({
  onToggle,
  address,
  headerContent,
  onClickSettings,
  collapsed = false,
  maxWidth = "304px",
  minWidth = "58px",
  onLogin,
  ...props
}) => {
  return (
    <Sidebar
      {...props}
      collapsed={collapsed}
      maxWidth={maxWidth}
      minWidth={minWidth}
    >
      <SidebarPanel>
        <ConnectSidebarHeader onToggle={onToggle}>
          {headerContent}
        </ConnectSidebarHeader>
        <div className="hidden expanded:block">{props.children}</div>
      </SidebarPanel>
      <ConnectSidebarFooter
        address={address}
        onClickSettings={onClickSettings}
        onLogin={onLogin}
      />
    </Sidebar>
  );
};
