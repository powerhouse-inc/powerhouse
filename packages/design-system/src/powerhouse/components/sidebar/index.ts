export * from "./sidebar";
export * from "./sidebar-panel";
