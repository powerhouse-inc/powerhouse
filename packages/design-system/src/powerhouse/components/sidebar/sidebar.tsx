import { useAnimation } from "@/powerhouse";
import { useEffect, useRef, useState } from "react";
import { twMerge } from "tailwind-merge";

export interface SidebarProps extends React.HTMLAttributes<HTMLDivElement> {
  collapsed: boolean;
  maxWidth: string;
  minWidth: string;
}

export type SidebarHeaderProps = React.HTMLAttributes<HTMLDivElement>;

export function SidebarHeader({ className, ...props }: SidebarHeaderProps) {
  return <div className={twMerge("shrink-0", className)} {...props} />;
}

export type SidebarFooterProps = React.HTMLAttributes<HTMLDivElement>;

export function SidebarFooter({ className, ...props }: SidebarFooterProps) {
  return <div className={twMerge("shrink-0", className)} {...props} />;
}

type AnimationState = "collapsed" | "expanded" | "collapsing" | "expanding";

export const Sidebar: React.FC<SidebarProps> = ({
  collapsed = false,
  maxWidth = "304px",
  minWidth = "80px",
  className,
  children,
  ...props
}) => {
  const ref = useRef<HTMLDivElement>(null);
  const [state, setState] = useState<AnimationState>(
    collapsed ? "collapsed" : "expanded",
  );

  useAnimation(ref, ["collapsing", "expanding"].includes(state), () => {
    setState((currentState) =>
      currentState === "collapsing"
        ? "collapsed"
        : currentState === "expanding"
          ? "expanded"
          : currentState,
    );
  });

  useEffect(() => {
    if (collapsed && state !== "collapsed") {
      setState("collapsing");
    } else if (!collapsed && state !== "expanded") {
      setState("expanding");
    }
  }, [collapsed]);

  return (
    <div
      {...props}
      className={twMerge(
        ` group ${state} flex
                 h-full flex-col overflow-hidden transition-none duration-300`,
        className,
        ["collapsing", "expanding"].includes(state) && "animate-out fade-out",
        ["collapsed", "expanded"].includes(state) && "animate-in fade-in",
        state === "collapsed" && "bg-transparent shadow-none",
        state === "expanded" && "shadow-sidebar",
      )}
      ref={ref}
      style={{
        width: ["collapsed", "expanding"].includes(state) ? minWidth : maxWidth,
      }}
    >
      {children}
    </div>
  );
};
