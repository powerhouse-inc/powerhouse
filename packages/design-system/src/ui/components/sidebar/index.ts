export * from "./sidebar.js";
export * from "./subcomponents/sidebar-provider/index.js";
export type * from "./types.js";
