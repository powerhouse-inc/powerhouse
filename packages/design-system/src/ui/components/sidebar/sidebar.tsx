"use client";

import { Icon } from "#powerhouse";
import { cn } from "#scalars";
import { useCallback, useEffect } from "react";
import { SidebarContentArea } from "./subcomponents/sidebar-content-area.js";
import { SidebarHeader } from "./subcomponents/sidebar-header.js";
import { SidebarPinningArea } from "./subcomponents/sidebar-pinning-area.js";
import { useSidebar } from "./subcomponents/sidebar-provider/index.js";
import { SidebarSearch } from "./subcomponents/sidebar-search/index.js";
import { type SidebarNode } from "./types.js";
import { useSidebarResize } from "./use-sidebar-resize.js";
import { triggerEvent } from "./utils.js";

export interface SidebarProps {
  /**
   * The ID of the currently active node
   */
  activeNodeId?: string;
  /**
   * Callback function to update the active node
   */
  onActiveNodeChange?: (newNode: SidebarNode) => void;
  /**
   * The nodes to be displayed in the sidebar
   */
  nodes?: SidebarNode[];
  /**
   * Which level should be shown by default (i.e., level 1).
   * @default 1 It is 1-indexed.
   */
  defaultLevel?: number;
  /**
   * Enables the vertical resizing of the sidebar
   * @default true
   */
  resizable?: boolean;
  /**
   * Whether pinning functionality is enabled for nodes
   * @default true
   */
  allowPinning?: boolean;
  /**
   * Configurable levels (e.g., 4). If its zero, then no buttons are shown (default).
   * @default 0
   */
  enableMacros?: number;
  /**
   * Whether to display the search bar in the sidebar
   * @default true
   */
  showSearchBar?: boolean;
  /**
   * Whether to display the status filter button in the sidebar. This option is ignored if `showSearchBar` is `false`.
   * @default false
   */
  showStatusFilter?: boolean;
  /**
   * The title that appears at the top of the sidebar, providing context or a heading for the hierarchy displayed within it.
   */
  sidebarTitle?: string;
  /**
   * An optional icon that can be displayed next to the sidebar title or at the top of the sidebar.
   */
  sidebarIcon?: React.ReactNode;
  /**
   * An optional footer content that can be displayed at the bottom of the sidebar.
   */
  extraFooterContent?: React.ReactNode;
  /**
   * The initial width of the sidebar.
   * @default 300
   */
  initialWidth?: number;
  /**
   * The maximum width of the sidebar.
   */
  maxWidth?: number;
  /**
   * Whether to allow collapsing inactive nodes on click
   * @default false
   */
  allowCollapsingInactiveNodes?: boolean;
  /**
   * Optional className for the sidebar container
   */
  className?: string;
}

/**
 * Sidebar component that provides a collapsible and resizable navigation panel
 * with support for hierarchical data, search, pinning, and custom styling.
 */
export const Sidebar: React.FC<SidebarProps> = ({
  activeNodeId,
  onActiveNodeChange,
  nodes,
  sidebarIcon,
  sidebarTitle,
  defaultLevel = 1,
  enableMacros = 0,
  allowPinning = true,
  resizable = true,
  showSearchBar = true,
  showStatusFilter = false,
  extraFooterContent,
  initialWidth = 300,
  maxWidth,
  allowCollapsingInactiveNodes = false,
  className,
}) => {
  const {
    sidebarRef,
    startResizing,
    isResizing,
    isSidebarOpen,
    handleToggleSidebar,
  } = useSidebarResize({
    defaultWidth: initialWidth,
    minWidth: 220,
    maxWidth,
  });

  const {
    pinnedNodePath,
    setNodes,
    openLevel,
    togglePin,
    syncActiveNodeId,
    setActiveNodeChangeCallback,
  } = useSidebar();

  // Sync external nodes with internal state when provided
  useEffect(() => {
    if (nodes) {
      setNodes(nodes);
    }
  }, [nodes, setNodes]);

  // Initialize default expanded level on mount
  useEffect(() => {
    if (defaultLevel > 1) {
      openLevel(defaultLevel);
    }
    // openLevel is intentionally omitted from dependencies
    // to prevent infinite loop as it may change on each render
  }, [defaultLevel]);

  const handleActiveNodeChange = useCallback(
    (node: SidebarNode) => {
      onActiveNodeChange?.(node);
      triggerEvent("sidebar:change", { node }, sidebarRef.current);
    },
    [onActiveNodeChange, sidebarRef],
  );
  // sync activeNodeId and onActiveNodeChange with provider state
  useEffect(() => {
    syncActiveNodeId(activeNodeId);
  }, [activeNodeId, syncActiveNodeId]);

  // Sync active node change callback with provider state
  useEffect(() => {
    setActiveNodeChangeCallback(handleActiveNodeChange);
  }, [handleActiveNodeChange, setActiveNodeChangeCallback]);

  // Unpin nodes if pinning is disabled
  useEffect(() => {
    const hasPinnedNodes = pinnedNodePath.length > 0;
    if (!allowPinning && hasPinnedNodes) {
      const lastPinnedNodeId = pinnedNodePath[pinnedNodePath.length - 1].id;
      togglePin(lastPinnedNodeId);
    }
  }, [allowPinning, pinnedNodePath, togglePin]);

  return (
    <aside
      ref={sidebarRef}
      style={{ width: "min(100%, var(--sidebar-width))" }}
      className={cn(
        "group peer relative flex h-svh max-h-screen flex-col bg-gray-50 shadow-lg transition-[width] duration-75 ease-linear dark:bg-slate-600",
        isResizing && "transition-none",
        className,
      )}
    >
      {isSidebarOpen && (
        <>
          <SidebarHeader
            sidebarTitle={sidebarTitle}
            sidebarIcon={sidebarIcon}
            enableMacros={enableMacros}
          />

          {allowPinning && pinnedNodePath.length > 0 && <SidebarPinningArea />}
          <SidebarContentArea
            allowPinning={allowPinning}
            allowCollapsingInactiveNodes={allowCollapsingInactiveNodes}
          />
          {showSearchBar && (
            <SidebarSearch showStatusFilter={showStatusFilter} />
          )}
          {extraFooterContent && (
            <div className="w-full border-t border-gray-300 p-2 dark:border-gray-800">
              {extraFooterContent}
            </div>
          )}
        </>
      )}

      {resizable && (
        <div
          className="group/sidebar-resizer absolute right-0 top-0 h-full w-[10px] translate-x-1/2 cursor-ew-resize select-none"
          onMouseDown={startResizing}
        >
          <div
            className={cn(
              "relative h-full w-px translate-x-[5px] transition-colors group-hover/sidebar-resizer:bg-gray-500 dark:group-hover/sidebar-resizer:bg-gray-600",
              isResizing && "cursor-default bg-blue-500",
              !isSidebarOpen && "bg-gray-300 dark:bg-gray-600",
            )}
          >
            <button
              type="button"
              className={cn(
                "absolute right-0 top-14 size-4 translate-x-1/2 rounded-full bg-gray-500 dark:bg-gray-900",
                "opacity-0 transition-opacity group-hover/sidebar-resizer:opacity-100",
              )}
              onClick={handleToggleSidebar}
            >
              {/* eslint-disable-next-line react/jsx-max-depth */}
              <Icon
                name="Caret"
                size={16}
                className={cn("min-w-4 text-gray-50 dark:text-gray-500", {
                  "-rotate-180": isSidebarOpen,
                })}
              />
            </button>
          </div>
        </div>
      )}
    </aside>
  );
};
