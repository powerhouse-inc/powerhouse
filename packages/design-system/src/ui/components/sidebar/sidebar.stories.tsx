import { Icon } from "#powerhouse";
import type { Meta, StoryObj } from "@storybook/react";
import { useCallback, useState } from "react";
import mockedTree from "./mocked_tree.json";
import { Sidebar } from "./sidebar.js";
import { SidebarProvider } from "./subcomponents/sidebar-provider/index.js";
import { type SidebarNode } from "./types.js";

/**
 * The `Sidebar` component can be used within a page layout to provide a sidebar navigation.
 * It provided a tree structure of nodes that can be used to navigate the application offering customization, search and more.
 *
 * The `Sidebar` component requires a `SidebarProvider` ancestor component to function correctly. The `SidebarProvider` accepts an optional `nodes` prop
 * which defines the navigation tree structure. If `nodes` is not provided to the provider, the `Sidebar` component can accept its own `nodes` prop.
 * If neither source provides nodes, the sidebar will render empty.
 *
 * Usage:
 *
 * ```tsx
 * <SidebarProvider nodes={nodes}>
 *   <Sidebar {...sidebarProps} />
 * </SidebarProvider>
 * ```
 *
 * The sidebar nodes are defined as follows:
 *
 * ```tsx
 * type SidebarNode = {
 *   id: string;
 *   title: string;
 *   children: SidebarNode[];
 *   icon?: IconName | ReactElement;
 *   expandedIcon?: IconName | ReactElement;
 *   status?: NodeStatus;
 * };
 * enum NodeStatus {
 *   CREATED = "CREATED",
 *   MODIFIED = "MODIFIED",
 *   REMOVED = "REMOVED",
 *   MOVED = "MOVED",
 *   DUPLICATED = "DUPLICATED",
 *   UNCHANGED = "UNCHANGED", // default status, no need to set it
 * }
 * ```
 *
 * The `icon` and `expandedIcon` properties are optional and can be used to display an icon in the sidebar item.
 * This icons must be one of the [available icons](?path=/docs/powerhouse-iconography--readme)
 *
 * ## Sidebar Events
 *
 * The `Sidebar` component emits the following custom events:
 *
 * - `sidebar:change`: it is triggered when the sidebar item is clicked.
 *  - Data: `{ node: SidebarNode }`
 * - `sidebar:resize:start`: it is triggered when the sidebar resize starts at the moment the user clicks down in the resizing handle.
 *  - Data: `{ isSidebarOpen: boolean }`
 * - `sidebar:resize:active`: it is triggered when the sidebar is being resized while the user is dragging the resizing handle.
 * It could be triggered multiple times while the user is dragging the resizing handle.
 *  - Data: `{ isSidebarOpen: boolean, sidebarWidth: number }`
 * - `sidebar:resize`: it is triggered when the sidebar resize stops at the moment the user releases the resizing handle.
 *  - Data: `{ isSidebarOpen: boolean, sidebarWidth: number }`
 * - `sidebar:resize:toggle`: it is triggered when the sidebar is toggled (collapsed or expanded).
 *  - Data: `{ isSidebarOpen: boolean }`
 *
 * ### Example of listening to the events
 *
 * ```tsx
 * useEffect(() => {
 *   const onResize = (event: Event) => {
 *     console.log("sidebar:resize", event);
 *   };
 *
 *   // you can add the listener directly to the document or add a
 *   // `className`, get the sidebar element and add the listener to it.
 *   document.addEventListener("sidebar:resize", onResize);
 *
 *   return () => {
 *     document.removeEventListener("sidebar:resize", onResize);
 *   };
 * }, []);
 * ```
 */
const meta: Meta<typeof Sidebar> = {
  title: "Document Engineering/Navigation/Sidebar",
  component: Sidebar,
  tags: ["autodocs"],
  decorators: [
    (Story) => (
      <SidebarProvider nodes={mockedTree as SidebarNode[]}>
        <Story />
      </SidebarProvider>
    ),
  ],
  parameters: {
    layout: "fullscreen",
  },
  argTypes: {
    activeNodeId: {
      control: "text",
      description: "The id of the node that is currently active.",
    },
    nodes: {
      control: "object",
      table: {
        readonly: true,
      },
      description:
        "The nodes to be displayed in the sidebar. It can be provided through the SidebarProvider to prevent flickering.",
    },
    sidebarTitle: {
      control: "text",
      description: "The title of the sidebar.",
    },
    sidebarIcon: {
      control: "object",
      table: {
        readonly: true,
      },
      description: "The icon of the sidebar.",
    },
    defaultLevel: {
      control: "number",
      description: "The level to be opened by default.",
      table: {
        defaultValue: { summary: "1" },
      },
    },
    enableMacros: {
      control: "number",
      description:
        "The number of macros to be displayed in the sidebar. Recommended up to 4.",
      table: {
        defaultValue: { summary: "0" },
      },
    },
    allowPinning: {
      control: "boolean",
      description: "Whether the sidebar items can be pinned.",
      table: {
        defaultValue: { summary: "true" },
      },
    },
    resizable: {
      control: "boolean",
      description: "Whether the sidebar is resizable.",
      table: {
        defaultValue: { summary: "true" },
      },
    },
    showSearchBar: {
      control: "boolean",
      description: "Whether the sidebar allows searching.",
      table: {
        defaultValue: { summary: "true" },
      },
    },
    showStatusFilter: {
      control: "boolean",
      description: "Whether the sidebar allows filtering by status.",
      table: {
        defaultValue: { summary: "false" },
      },
    },
    extraFooterContent: {
      control: "object",
      description: "Additional content to be displayed in the sidebar footer.",
      table: {
        readonly: true,
      },
    },
    initialWidth: {
      control: "number",
      description: "The initial width of the sidebar.",
      table: {
        defaultValue: { summary: "300" },
      },
    },
    maxWidth: {
      control: "number",
      description: "The maximum width of the sidebar.",
    },
    allowCollapsingInactiveNodes: {
      control: "boolean",
      description: "Whether to allow collapsing inactive nodes on click.",
      table: {
        defaultValue: { summary: "false" },
      },
    },
  },
  args: {
    sidebarTitle: "Title Sidebar",
    sidebarIcon: (
      <div className="flex items-center justify-center rounded-md bg-gray-900 p-2">
        <Icon name="M" className="text-gray-50" size={16} />
      </div>
    ),
    enableMacros: 4,
    onActiveNodeChange: (node) => {
      console.log("onActiveNodeChange", node);
    },
  },
};

export default meta;
type Story = StoryObj<typeof Sidebar>;

export const Default: Story = {};

/**
 * The `Sidebar` component can be used within a page layout to provide a sidebar navigation.
 * It can also be used to display additional content within the sidebar, such as authentication buttons.
 * or any other react component.
 */
export const WithinLayoutAndContent: Story = {
  args: {
    showStatusFilter: true,
  },
  render: (args) => {
    const [activeNode, setActiveNode] = useState<string>(
      "4281ab93-ef4f-4974-988d-7dad149a693d",
    );

    const onActiveNodeChange = useCallback((node: SidebarNode) => {
      setActiveNode(node.id);
    }, []);

    return (
      <main className="flex h-svh w-full">
        <Sidebar
          className="sidebar"
          {...args}
          onActiveNodeChange={onActiveNodeChange}
          activeNodeId={activeNode}
          extraFooterContent={
            <div className="flex flex-col gap-1">
              <div className="flex items-baseline gap-2 text-gray-900 dark:text-gray-200">
                <div>Login with</div>
                <Icon
                  name={"Renown"}
                  size={"auto"}
                  height={18}
                  className="cursor-pointer"
                />
              </div>
              <div className="text-xs text-gray-500 dark:text-gray-400">
                Auth integration example within sidebar content
              </div>
            </div>
          }
        />
        <div
          style={{ width: "calc(100% - var(--sidebar-width))" }}
          className="flex-1 bg-gray-50 p-4 dark:bg-slate-800"
        >
          <h1 className="text-2xl font-bold text-gray-900 dark:text-gray-50">
            Content Area
          </h1>
          <p className="text-sm text-gray-500 dark:text-gray-400">
            Active Node: {activeNode}
          </p>
        </div>
      </main>
    );
  },
};
