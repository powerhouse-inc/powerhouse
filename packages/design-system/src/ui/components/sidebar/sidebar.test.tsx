import { Icon } from "#powerhouse";
import { render, screen } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import { Sidebar } from "./sidebar.js";
import { SidebarProvider } from "./subcomponents/sidebar-provider/index.js";

const mockNodes = [
  {
    id: "1",
    title: "Node 1",
    children: [
      {
        id: "1.1",
        title: "Node 1.1",
        children: [],
      },
    ],
  },
  {
    id: "2",
    title: "Node 2",
    children: [],
  },
];

const renderSidebar = (props = {}) => {
  return render(
    <SidebarProvider nodes={mockNodes}>
      <Sidebar {...props} />
    </SidebarProvider>,
  );
};

describe("Sidebar Component", () => {
  it("should match snapshot", () => {
    const { asFragment } = renderSidebar({
      sidebarTitle: "Test Sidebar",
      sidebarIcon: <Icon name="M" size={16} />,
      enableMacros: 2,
      defaultLevel: 1,
    });
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render sidebar title and icon when provided", () => {
    renderSidebar({
      sidebarTitle: "Test Sidebar",
      sidebarIcon: <div data-testid="sidebar-icon">Icon</div>,
    });

    expect(screen.getByText("Test Sidebar")).toBeInTheDocument();
    expect(screen.getByTestId("sidebar-icon")).toBeInTheDocument();
  });

  it("should render search bar by default", () => {
    renderSidebar();
    expect(screen.getByRole("searchbox")).toBeInTheDocument();
  });

  it("should not render search bar when showSearchBar is false", () => {
    renderSidebar({ showSearchBar: false });
    expect(screen.queryByRole("searchbox")).not.toBeInTheDocument();
  });

  it("should render extra footer content when provided", () => {
    renderSidebar({
      extraFooterContent: <div>Extra Footer Content</div>,
    });
    expect(screen.getByText("Extra Footer Content")).toBeInTheDocument();
  });

  it("should render pinning area when there are pinned items", () => {
    renderSidebar({
      allowPinning: true,
    });

    // Initially pinning area should not be visible
    expect(screen.queryByTestId("pinning-area")).not.toBeInTheDocument();
  });
});
