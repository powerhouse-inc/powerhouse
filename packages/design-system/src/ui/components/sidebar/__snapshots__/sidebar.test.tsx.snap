// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Sidebar Component > should match snapshot 1`] = `
<DocumentFragment>
  <aside
    class="group peer relative flex h-svh max-h-screen flex-col bg-gray-50 shadow-lg transition-[width] duration-75 ease-linear dark:bg-slate-600"
    style="width: min(100%, var(--sidebar-width));"
  >
    <header
      class="flex items-center justify-between gap-2 border-b border-gray-300 bg-gray-50 p-4 dark:border-gray-800 dark:bg-slate-600"
    >
      <div
        class="flex items-center gap-2 truncate"
      >
        <div
          data-testid="icon-fallback"
          style="width: 16px; height: 16px;"
        />
        <div
          class="truncate text-sm font-semibold text-gray-700 dark:text-gray-300"
        >
          Test Sidebar
        </div>
      </div>
      <div
        class="flex select-none items-center gap-2"
      >
        <div
          class="w-[26px] rounded-lg bg-slate-50 p-1 text-center text-xs text-slate-100 dark:bg-gray-900 dark:text-slate-200 hover:bg-slate-100 hover:text-slate-200 dark:hover:bg-gray-600 dark:hover:text-slate-50"
          role="button"
          tabindex="2"
        >
          1
        </div>
        <div
          class="w-[26px] rounded-lg bg-slate-50 p-1 text-center text-xs text-slate-100 dark:bg-gray-900 dark:text-slate-200 hover:bg-slate-100 hover:text-slate-200 dark:hover:bg-gray-600 dark:hover:text-slate-50"
          role="button"
          tabindex="2"
        >
          2
        </div>
      </div>
    </header>
    <div
      class="flex flex-1 flex-col gap-1 overflow-y-auto"
    >
      <div
        class="flex-1 overflow-hidden"
      >
        <div
          style="overflow: visible; height: 0px; width: 0px;"
        >
          <div
            aria-label="grid"
            aria-readonly="true"
            class="ReactVirtualized__Grid ReactVirtualized__List p-2"
            role="grid"
            style="box-sizing: border-box; direction: ltr; height: 0px; position: relative; width: 0px; will-change: transform; overflow-x: hidden; overflow-y: auto;"
            tabindex="0"
          />
        </div>
        <div
          class="resize-triggers"
        >
          <div
            class="expand-trigger"
          >
            <div
              style="width: 1px; height: 1px;"
            />
          </div>
          <div
            class="contract-trigger"
          />
        </div>
      </div>
    </div>
    <div
      class="flex w-full gap-2 border-t border-gray-300 p-2 dark:border-gray-800"
    >
      <div
        class="relative flex-1"
      >
        <input
          class="flex h-9 rounded-md text-sm font-normal leading-5 text-gray-900 dark:text-gray-50 dark:border-charcoal-700 dark:bg-charcoal-900 border border-gray-300 bg-white px-3 py-2 font-sans placeholder:text-gray-500 dark:placeholder:text-gray-600 focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-gray-900 focus-visible:ring-offset-0 focus-visible:ring-offset-white dark:focus-visible:ring-charcoal-300 dark:focus-visible:ring-offset-charcoal-900 dark:focus:bg-charcoal-900 focus:bg-gray-50 disabled:cursor-not-allowed disabled:border-gray-300 disabled:bg-white disabled:text-gray-700 disabled:dark:border-charcoal-800 disabled:dark:bg-charcoal-900 disabled:dark:text-gray-300 w-full appearance-none !pl-8 [&::-webkit-search-cancel-button]:hidden"
          placeholder="Search"
          style="padding-right: 16px;"
          tabindex="1"
          type="search"
          value=""
        />
        <svg
          class="absolute left-2 top-1/2 -translate-y-1/2 text-gray-700 dark:text-gray-50"
          fill="currentcolor"
          height="16"
          viewBox="0 0 24 24"
          width="16"
        >
          <path
            d="M10.0137 1.95312C5.59537 1.95312 2.01367 5.53513 2.01367 9.95314C2.01367 14.3712 5.59537 17.9532 10.0137 17.9532C11.8504 17.9532 13.5913 17.3222 14.9416 16.2822L19.2949 20.6722C19.6855 21.0622 20.3419 21.0622 20.7325 20.6722C21.123 20.2812 21.123 19.6252 20.7325 19.2342L16.349 14.8741C17.3898 13.5241 18.0137 11.7901 18.0137 9.95314C18.0137 5.53513 14.432 1.95312 10.0137 1.95312ZM10.0137 3.95313C13.3274 3.95313 16.0137 6.63914 16.0137 9.95314C16.0137 13.2672 13.3274 15.9532 10.0137 15.9532C6.69997 15.9532 4.01367 13.2672 4.01367 9.95314C4.01367 6.63914 6.69997 3.95313 10.0137 3.95313Z"
          />
        </svg>
      </div>
    </div>
    <div
      class="group/sidebar-resizer absolute right-0 top-0 h-full w-[10px] translate-x-1/2 cursor-ew-resize select-none"
    >
      <div
        class="relative h-full w-px translate-x-[5px] transition-colors group-hover/sidebar-resizer:bg-gray-500 dark:group-hover/sidebar-resizer:bg-gray-600"
      >
        <button
          class="absolute right-0 top-14 size-4 translate-x-1/2 rounded-full bg-gray-500 dark:bg-gray-900 opacity-0 transition-opacity group-hover/sidebar-resizer:opacity-100"
          type="button"
        >
          <div
            data-testid="icon-fallback"
            style="width: 16px; height: 16px;"
          />
        </button>
      </div>
    </div>
  </aside>
</DocumentFragment>
`;
